crypto/bio/bio.c

```diff
@@ -734,15 +734,57 @@ int (*BIO_meth_get_destroy(const BIO_METHOD *method)) (BIO *) {
   return method->destroy;
 }
 
+static int bwrite_ex_to_briwte_conversion(BIO *bio, const char *input, int requested) {
+  if (requested > INT_MAX) {
+    requested = INT_MAX;
+  }
+  size_t actual = 0;
+  const int success = bio->method->bwrite_ex(bio, input, requested, &actual);
+  if (success <= 0) {
+    return success;
+  }
+  // this is safe because we capped the write to INT_MAX
```

Comment:
This implies the `actual` value can never be more than `requested`. This is a reasonable expectation, but it's unenforceable. We can't prevent the callback from setting the value of `actual` to a larger value.